packages/button/src/Button.tsx

```diff
@@ -1,483 +1,182 @@
-  background-color: ${uiColors.gray.light2};
-  background-image: none;
-  box-shadow: none;
-  pointer-events: none;
-`;
 
-interface BaseButtonProps {
-  disabled?: boolean;
-  darkMode?: boolean;
-  variant?: Variant;
-  size?: Size;
-  glyph?: React.ReactElement;
-  className?: string;
-  borderRadius?: string;
-  children?: React.ReactNode;
-  href?: string;
-  forceState?: {
-    focused?: boolean;
-    active?: boolean;
-    // Supporting "hovered" requires factoring out all the &:hover
-    // styles from the the base style and each variant's styles
+  const buttonProps = {
+    type,
+    className: cx(buttonClassName, className),
+    ref: forwardRef,
+    // Provide a default value for the as prop
+    // If consumping application passes a value for as, it will override the default set here
+    as: isAnchor ? 'a' : 'button',
+    // only add a disabled prop if not an anchor
+    ...(typeof rest.href !== 'string' && { disabled }),
+    'aria-disabled': disabled,
+    ...rest,
+  } as const;
+
+  const accessibleIconProps = !isIconOnlyButton && {
+    'aria-hidden': true,
+    role: 'presentation',
   };
-}
-
-const Button: ExtendableBox<
-  BaseButtonProps & { ref?: React.Ref<any> },
-  'button'
-> = React.forwardRef(
-  (
-    {
-      className,
-      borderRadius,
-      children,
-      disabled = false,
-      darkMode = false,
-      variant = Variant.Default,
-      size = Size.Normal,
-      glyph,
-      forceState = {},
-      ...rest
-    }: BaseButtonProps,
-    ref: React.Ref<any>,
-  ) => {
-    const { usingKeyboard: showFocus } = useUsingKeyboardContext();
 
-    const commonProps = {
-      ref,
+  const clonedLeftGlyph =
+    leftGlyph &&
+    React.cloneElement(leftGlyph, {
       className: cx(
-        baseStyle,
-        buttonSizes[size],
-        buttonVariants[variant],
-        { [disabledStyle]: disabled },
-        { [focusStyle]: showFocus },
         {
           [css`
-            border-radius: ${borderRadius};
-
-            &:before,
-            &:after {
-              border-radius: calc(${borderRadius} - 1px);
-            }
-          `]: borderRadius !== undefined,
-        },
-        {
-          [css`
-            &:before {
-              opacity: 1;
-            }
-          `]: forceState.focused === true,
+            margin-right: ${iconSpacing[size]}px;
+          `]: !isIconOnlyButton || !!rightGlyph,
         },
+        getIconStyle({ variant, size, darkMode, disabled, isIconOnlyButton }),
+      ),
+      ...accessibleIconProps,
+    });
+
+  const clonedRightGlyph =
+    rightGlyph &&
+    React.cloneElement(rightGlyph, {
+      className: cx(
         {
           [css`
-            &:after {
-              opacity: 1;
-            }
-          `]: forceState.active === true,
+            margin-left: ${iconSpacing[size]}px;
+          `]: !isIconOnlyButton || !!leftGlyph,
         },
+        getIconStyle({ variant, size, darkMode, disabled, isIconOnlyButton }),
```

Comment:
I have very mixed feelings about this sort of function. To me, it looks like we should be building a component to handle this since we're effectively handling props as-is.

Reply:
What do you mean by building a component to handle this?

Reply:
I actually think that this makes it really clear what styles are being applied to the button and why (the function itself is very readable) as opposed to having a really messily composed styleset where you can't tell what prop is effecting what behavior

Reply:
It's very unclear what you're doing to the icon if you're trying to read `Button.tsx` directly. By moving this logic to a component, you can make all that logic related to cloning the glyph and such self-contained, and co-locate styles related to the glyphs with the glyph implementation itself while still keeping `Button.tsx from getting bloated.

Reply:
Created an Icon component, let me know what you think